src/init.cpp

```diff
@@ -954,10 +954,14 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     if (mapArgs.count("-minrelaytxfee"))
     {
         CAmount n = 0;
-        if (ParseMoney(mapArgs["-minrelaytxfee"], n) && n > 0)
-            ::minRelayTxFee = CFeeRate(n);
-        else
+        if (!ParseMoney(mapArgs["-minrelaytxfee"], n))
             return InitError(AmountErrMsg("minrelaytxfee", mapArgs["-minrelaytxfee"]));
+#ifdef ENABLE_WALLET
+        if (!fDisableWallet && n > HIGH_TX_FEE_PER_KB)
+            InitWarning(AmountHighWarn("-minrelaytxfee") + " " +
+                        _("This is the minimum transaction fee you pay on every transaction."));
```

Comment:
Right, but this is irrelevant for the wallet. If you run with the wallet disabled, you can use minrelayfee with the feefilter to approach a `-blocksonly` mode for minrelayfee → ∞.

With the wallet enabled, it should warn when you are about to pay a fee too high.

I have moved this code to wallet.cpp to make that clearer.